src/PostgREST/Query/Statements.hs

```diff
@@ -49,15 +49,19 @@ data ResultSet
   -- ^ the HTTP headers to be added to the response
   , rsGucStatus  :: Maybe Text
   -- ^ the HTTP status to be added to the response
+  , rsInserted   :: Maybe Int64
+  -- ^ the number of rows inserted (Only used with PUT)
   }
   | RSPlan BS.ByteString -- ^ the plan of the query
 
 
-prepareWrite :: SQL.Snippet -> SQL.Snippet -> Bool -> MediaType -> ResultAggregate ->
+prepareWrite :: SQL.Snippet -> SQL.Snippet -> Bool -> Bool -> MediaType -> ResultAggregate ->
                 Maybe PreferRepresentation -> [Text] -> Bool -> SQL.Statement () ResultSet
-prepareWrite selectQuery mutateQuery isInsert mt rAgg rep pKeys =
+prepareWrite selectQuery mutateQuery isInsert isPut mt rAgg rep pKeys =
   SQL.dynamicallyParameterized (mtSnippet mt snippet) decodeIt
  where
+  checkUpsert snip = if isInsert && isPut then snip else "''"
+  pgrstInsertedF = checkUpsert "current_setting('pgrst.inserted', true)"
```

Comment:
This needs to be converted to an `int`. Also needs the wrap with `nullif` or `''::int` may give errors.

```suggestion
  pgrstInsertedF = checkUpsert "nullif(current_setting('pgrst.inserted', true),'')::int"
```

Or maybe you could take it as a boolean (and change `rsInserted` to `Maybe Bool`).

```suggestion
  pgrstInsertedF = checkUpsert "current_setting('pgrst.inserted', true) = '1"
```

Reply:
> This needs to be converted to an `int`. Also needs the wrap with `nullif` or `''::int` may give errors.

Hmm, I have made these changes, however `PUT` is still not responding with `201` on inserts. It seems very strange since I can't figure out anything wrong in the implementation.

Reply:
Ah, this should be inside `singleUpsertResponse` instead of `updateResponse`.